File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] prefix=new int[nums.length];
        int[] sufix=new int[nums.length];
        for(int i=0;i<nums.length;i++){
            if(i==0){
                prefix[i]=1;
                continue;
            }
            prefix[i]=prefix[i-1]*nums[i-1];
        }
        for(int j=nums.length-1;j>=0;j--){
            if(j==nums.length-1){
                sufix[j]=1;
                continue;
            }
            sufix[j]=sufix[j+1]*nums[j+1];
        }
        for(int i=0;i<nums.length;i++){
            nums[i]=prefix[i]*sufix[i];
        }
        return nums;
    }
}  

File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res=[]
        subset=[]
        def dfs(i,curr_sum):
            if curr_sum==target:
                res.append(subset[:])
                return 
            if curr_sum>target:
                return
            if i>=len(nums):
                return
            subset.append(nums[i])
            curr_sum+=nums[i]
            dfs(i,curr_sum)
            remove=subset.pop()
            curr_sum-=remove
            dfs(i+1,curr_sum)
        dfs(0,0)
        return res            